pkg/3scale/amp/operator/secret_reconciler.go

```diff
@@ -3,88 +3,79 @@ package operator
 import (
 	"context"
 	"fmt"
-	"reflect"
 
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-type SecretReconciler struct {
-	BaseReconciler
-	ObjectMetaMerger ObjectMetaMerger
+type SecretReconciler interface {
+	IsUpdateNeeded(desired, existing *v1.Secret) bool
 }
 
-func NewSecretReconciler(baseReconciler BaseReconciler, objectMetaMerger ObjectMetaMerger) SecretReconciler {
-	return SecretReconciler{
-		BaseReconciler:   baseReconciler,
-		ObjectMetaMerger: objectMetaMerger,
+type SecretBaseReconciler struct {
+	BaseAPIManagerLogicReconciler
+	reconciler SecretReconciler
+}
+
+func NewSecretBaseReconciler(baseAPIManagerLogicReconciler BaseAPIManagerLogicReconciler, reconciler SecretReconciler) *SecretBaseReconciler {
+	return &SecretBaseReconciler{
+		BaseAPIManagerLogicReconciler: baseAPIManagerLogicReconciler,
+		reconciler:                    reconciler,
 	}
 }
 
-// TODO should this be a shared Secret reconcile behaviour or could there be
-// different secrets where we reconcile them in a different way?
-// For example, secrets where we just want to check if they are installed it
-// but we don't want to reconcile them. That would make sense for example
-// for system-seed secret.
-// Also in the case where we wanted to reconcile them but we wanted to do it
-// in a different way
-func (r *SecretReconciler) Reconcile(desiredSecret *v1.Secret) error {
+func (r *SecretBaseReconciler) Reconcile(desiredSecret *v1.Secret) error {
```

Comment:
With the changes we have lost the reconciliation of Secret defaults to allow the users to set specific desired values  for them before deploying APIManager. We should allow that.
Notice that I'm not saying reconciling secrets after APIManager has been deployed, which at this moment doesn't work and didn't work either automatically out of the box due to DeploymentConfigs not reacting to that.